bubblesub/cmd/show_widget.py

```diff
@@ -0,0 +1,71 @@
+# bubblesub - ASS subtitle editor
+# Copyright (C) 2018 Marcin Kurczewski
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import argparse
+import enum
+
+from PyQt5 import QtWidgets
+
+from bubblesub.api import Api
+from bubblesub.api.cmd import BaseCommand
+
+
+class _TargetWidget(enum.Enum):
+    """Known widgets in GUI."""
+
+    def __str__(self) -> str:
+        return self.value
+
+    NoteEditor = "note-editor"
+    Console = "console"
+
+
+class ShowWidgetCommand(BaseCommand):
+    names = ["show-widget"]
+    help_text = "Shows given widget."
+
+    async def run(self) -> None:
+        await self.api.gui.exec(self._run_with_gui)
+
+    async def _run_with_gui(self, main_window: QtWidgets.QMainWindow) -> None:
+        widget = main_window.findChild(
+            QtWidgets.QWidget, str(self.args.target),
+        )
+
+        if self.args.all:
+            widget = widget.parentWidget()
+
+        widget.setEnabled(not widget.isEnabled())
+        widget.setVisible(not widget.isVisible())
+
+    @staticmethod
+    def decorate_parser(api: Api, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            "target",
+            help="which widget to show/hide",
+            type=_TargetWidget,
+            choices=list(_TargetWidget),
+        )
+
+        parser.add_argument(
+            "-a",
+            "--all",
+            help="consider every children of a widget",
+            action="store_true",
+        )
```

Comment:
I'd like to see `BooleanOperation` argument here to decide whether to toggle, enable or disable the given widget (look in `pause.py`)

Reply:
So should I replace this option with a boolean that enables/disables a widget? Since you said more forward in the review that you don't want to disable a widget, I think we could remove this option.

Reply:
Yes IMO we should get rid of the `setDisabled` altogether unless you have a specific reason why you did this

Reply:
Nope, it was a copy-paste error. If you want we could disable a widget to impede the user to interact with it, but I don't know if it's the right approach.